test/neighbor_searcher_test.cpp

```diff
@@ -0,0 +1,52 @@
+#include "domain.hpp"
+#include "neighbor_searcher.hpp"
+
+#include <gtest/gtest.h>
+#include <random>
+#include <set>
+
+TEST(NeighborSearcherTest, NeighborSearch2d) {
+    double re           = 0.1;
+    double re2          = re * re;
+    size_t particleSize = 100;
+    Domain domain;
+    domain.xMin = 0.0;
+    domain.xMax = 1.0;
+    domain.yMin = 0.0;
+    domain.yMax = 2.0;
+    domain.zMin = 0.0;
+    domain.zMax = 0.0;
+    domain.xLength = domain.xMax - domain.xMin;
+    domain.yLength = domain.yMax - domain.yMin;
+    domain.zLength = domain.zMax - domain.zMin;
+
+    std::default_random_engine engine(0);
+    std::uniform_real_distribution<double> dist_x(domain.xMin, domain.xMax);
+    std::uniform_real_distribution<double> dist_y(domain.yMin, domain.yMax);
+
+    auto p = Particles();
+    for (size_t i = 0; i < particleSize; i++) {
+        auto r_i = Eigen::Vector3d(dist_x(engine), dist_y(engine), 0.0);
+        auto u_i = Eigen::Vector3d::Zero();
+        p.add(Particle(i, ParticleType::Fluid, r_i, u_i, 1.0, 0));
+    }
+
+    NeighborSearcher searcher(re, domain, particleSize);
+    searcher.setNeighbors(p);
+    for (const auto& pi : p) {
+        std::set<int> neighbors_bf;
+        for (const auto& pj : p) {
+            if (pi.id == pj.id) {
+                continue;
+            }
+            if ((pi.position - pj.position).squaredNorm() < re2) {
+                neighbors_bf.insert(pj.id);
+            }
+        }
+        std::set<int> neighbors_seach;
```

Comment:
```suggestion
        std::set<int> neighbors_search;
```

ですかね
命名規則的には neighborsSearch のほうが良さそう

また，命名としては

neighborsByBluteForce, neighborsBySearcher

みたいな対比だと読みやすいかもです

Reply:
55fc66a981e7704cb095f241b788af4c9bd14cc5 でneighborsByBluteForce, neighborsBySearcherに修正しました